examples/simple/simple.cpp

```diff
@@ -0,0 +1,208 @@
+    struct ggml_tensor* result = ggml_mul_mat(ctx0, model.a, model.b);
+
+    ggml_build_forward_expand(gf, result);
+
+    // delete the temporally context used to build the graph
+    ggml_free(ctx0);
+    return gf;
+}
+
+// compute with backend
+struct ggml_tensor* compute(const simple_model & model, struct ggml_allocr * allocr) {
+    // reset the allocator to free all the memory allocated during the previous inference
+    ggml_allocr_reset(allocr);
+
+    struct ggml_cgraph * gf = build_graph(model, allocr);
+
+    // allocate tensors
+    ggml_allocr_alloc_graph(allocr, gf);
+
+    int n_threads = 1; // number of threads to perform some operations with multi-threading
+
+    if (ggml_backend_is_cpu(model.backend)) {
+        ggml_backend_cpu_set_n_threads(model.backend, n_threads);
+    }
+
+#ifdef GGML_USE_METAL
+    if (ggml_backend_is_metal(model.backend)) {
+        ggml_backend_metal_set_n_cb(model.backend, n_threads);
+    }
+#endif
+
+    ggml_backend_graph_compute(model.backend, gf);
+
+    // in this case, the output tensor is the last one in the graph
+    return gf->nodes[gf->n_nodes - 1];
+}
+
+int main(void)
+{
+    ggml_time_init();
+
+    // initialize data of matrices to perfom matrix multiplication
+
+    float matrix_A[4] = {
+        2, 8,
+        5, 1
+    };
+
+    // transpose of [10, 5, 9, 9]
+    float matrix_B[4] = {
+        10, 9,
+        5, 9
+    };
+
+    simple_model model;
+    load_model(model, matrix_A, matrix_B);
+
+    // calculate the temporaly memory required to compute
+    ggml_backend_buffer_t buf_compute;
+    struct ggml_allocr * allocr = NULL;
+
+    {
+        size_t align = ggml_backend_get_alignment(model.backend);
+        allocr = ggml_allocr_new_measure(align);
+
+        //create the worst case graph for memory usage estimation
+        struct ggml_cgraph * gf = build_graph(model, allocr);
+        size_t mem_size = ggml_allocr_alloc_graph(allocr, gf);
+        ggml_allocr_free(allocr);
+
+        // compute the required memory
+        buf_compute = ggml_backend_alloc_buffer(model.backend, mem_size);
+        allocr = ggml_allocr_new_from_buffer(buf_compute);
+        fprintf(stderr, "%s: compute buffer size: %.4f KB\n", __func__, mem_size/1024.0);
+    }
+
+    // perform computation
+    struct ggml_tensor * result = compute(model, allocr);
+
+    // create a array to print result
+    float* out_data = new float[2 * 2];
+
+    // bring the data from the backend memory
+    ggml_backend_tensor_get(result, out_data, 0, ggml_nbytes(result));
+
+    // expected result: 92.00 59.00 82.00 34.00
+    printf("mult mat:\n%.2f %.2f %.2f %.2f\n",
+                out_data[0], out_data[1], out_data[2], out_data[3]);
+
+    // release backend memory used for computation
+    ggml_backend_buffer_free(buf_compute);
+
+    // free memory
+    ggml_free(model.ctx);
+
+    // release backend memory and free backend
+    ggml_backend_buffer_free(model.buffer);
+    ggml_backend_free(model.backend);
```

Comment:
The `allocr` should also be freed.

Reply:
It's also possible to use `ggml_allocr_new_from_backend` instead of allocating a buffer separately. Then you only have the `allocr` object to free.